src/app/public/modules/split-view/split-view.component.ts

```diff
@@ -0,0 +1,375 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  animate,
+  style,
+  transition,
+  trigger
+} from '@angular/animations';
+
+import {
+  SkyAppWindowRef,
+  SkyCoreAdapterService,
+  SkyMediaBreakpoints,
+  SkyMediaQueryService
+} from '@skyux/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  Subscription
+} from 'rxjs/Subscription';
+
+import 'rxjs/add/operator/take';
+
+import 'rxjs/add/operator/takeUntil';
+
+import 'rxjs/add/operator/takeWhile';
+
+import 'rxjs/add/observable/fromEvent';
+
+import {
+  SkySplitViewBeforeWorkspaceCloseHandler
+} from './types/split-view-before-workspace-close-handler';
+
+import {
+  SkySplitViewMessage
+} from './types/split-view-message';
+
+import {
+  SkySplitViewMessageType
+} from './types/split-view-message-type';
+
+import {
+  SkySplitViewMediaQueryService
+} from './split-view-media-query.service';
+
+import {
+  SkySplitViewDrawerComponent
+} from './split-view-drawer.component';
+
+import {
+  SkySplitViewWorkspaceComponent
+} from './split-view-workspace.component';
+
+let nextId = 0;
+
+@Component({
+  selector: 'sky-split-view',
+  templateUrl: './split-view.component.html',
+  styleUrls: ['./split-view.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger(
+      'blockAnimationOnLoad', [
+        transition(':enter', [])
+      ]
+    ),
+    trigger(
+      'drawerEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(-100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    ),
+    trigger(
+      'workspaceEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    )
+  ]
+})
+export class SkySplitViewComponent implements OnInit, AfterViewInit, OnDestroy {
+
+  @Input()
+  public backLabel: string;
+
+  @Input()
+  public set drawerWidth(value: number) {
+    if (value) {
+      this._drawerWidth = value;
+      this.updateBreakpoints();
+    }
+  }
+
+  public get drawerWidth() {
+    if (this.isMobile) {
+      return undefined;
+    } else {
+      if (this._drawerWidth > this.drawerWidthMax) {
+        return this.drawerWidthMax;
+      } else if (this._drawerWidth < this.drawerWidthMin) {
+        return this.drawerWidthMin;
+      } else {
+        return this._drawerWidth || this.drawerWidthDefault;
+      }
+    }
+  }
+
+  @Input()
+  public messageStream = new Subject<SkySplitViewMessage>();
+
+  @Output()
+  public beforeWorkspaceClose = new EventEmitter<SkySplitViewBeforeWorkspaceCloseHandler>();
```

Comment:
Do we still need this? It adds a lot of complexity just for the mobile view. Since we're assuming the developer will be handling all of the states and whether or not a form can be closed, the developer could focus the workspace with the message stream and launch an alert if the user opens the drawer and clicks on a link, if the form has unchanged work. Let me know if I'm missing a scenario.

Reply:
This is to prevent users from losing work when in responsive mode and clicking the "back to list" button.

Reply:
Correct. Can the drawer be shown overtop of the workspace, without removing the workspace from the DOM? If so, my description above should be a possible workaround.

Reply:
This has been removed.